File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
from collections import defaultdict
def solution(participant, completion):
    counter = defaultdict(int)
    
    for name in participant:
        counter[name] += 1
    
    for name in completion:
        counter[name] -= 1
    
    for name,count in counter.items():
        if count > 0:
            return name
    

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    const counter = new Map()
    
    for (const name of participant){
        counter.set(name, (counter.get(name) || 0) + 1)
    }
    
    for (const name of completion){
        counter.set(name, (counter.get(name) - 1))
    }
    
    for (const [name, count] of counter){
        if (count > 0) return name;
        
    }
}